src/action.c

```diff
@@ -166,9 +166,23 @@ show_menu(struct server *server, struct view *view, const char *menu_name)
 }
 
 static struct view *
-activator_or_focused_view(struct view *activator, struct server *server)
+view_for_action(struct view *activator, struct server *server,
+	struct action *action)
 {
-	return activator ? activator : desktop_focused_view(server);
+	/* View is explicitly specified for mousebinds */
+	if (activator) {
+		return activator;
+	}
+
+	/* Select view based on action type for keybinds */
+	switch (action->type) {
+	case ACTION_TYPE_FOCUS:
+	case ACTION_TYPE_MOVE:
+	case ACTION_TYPE_RESIZE:
+		return get_cursor_context(server).view;
+	default:
```

Comment:
Do we need `Raise` here as well?

Reply:
I don't personally use any non-default `Focus` and `Raise` actions though, so I'm probably not the target user here ... that said, I would vote for keeping the existing behavior (where `Raise` applies to the focused view), or open a new issue/pull request if there's a good reason to change it.

Reply:
>  I don't personally use any non-default Focus and Raise actions

Same for me. I would actually like to get rid of those two actions completely and let labwc handle it internally.
Even Openbox warns about custom mouse bindings. But that is a discussion for another time.